src/storage/mvcc/txn.rs

```diff
@@ -250,6 +261,12 @@ impl<S: Snapshot> MvccTxn<S> {
                     }
                     None => {
                         let ts = self.start_ts;
+
+                        // collapse previous rollback if exist.
+                        if self.collapse_rollback {
```

Comment:
please use TLA+ to verify it @cwen0

Reply:
I have used TLA+ to prove that the implementation of pr is ok , ref to https://github.com/pingcap/tla-plus/pull/23

Reply:
@GregoryIan any problem?

Reply:
I have asked @GregoryIan, and he said there is no problem.